src/server/util/books/kakao.js

```diff
@@ -0,0 +1,171 @@
+        return KakaoBookAPIUtil.searchBooks(keyword, {pageNo})
+          .then(data => data.bookList)
+          .then(list => {
+            return list.filter(data2 => data2.isbn);
+          })
+      }));
+    });
+  }
+
+  static async searchAllBooks(keyword) {
+    let hasNext = true;
+    let pageNo = 1;
+    let bookList = [];
+    bookList.push([]);
+    while (hasNext) {
+      const result = await KakaoBookAPIUtil.searchBooks(keyword, {pageNo});
+      bookList.push(result.bookList);
+      pageNo = result.pageInfo.pageNo + 1;
+      hasNext = result.pageInfo.hasNext;
+    }
+    return bookList.reduce((prev, next) => prev.concat(next));
+  }
+
+  static responseParser(res, pageNo, result) {
+    const searchResult = {};
+    searchResult.pageInfo = KakaoBookAPIUtil.pageParser(res, pageNo, result);
+    searchResult.bookList = KakaoBookAPIUtil.bookListParser(res);
+    return searchResult;
+  }
+
+  static bookListParser(data) {
+    if (data.channel.item.length === 0) return [];
+    return data.channel.item.map(KakaoBookAPIUtil.bookParser);
+  }
+
+  static bookParser(item) {
+    /*
+    example of part of response item from kakao api
+    {
+      "author": String,     // "진중권 김기승 외 1 명"m
+      "cover_l_url": String,// "http70111090716"
+      "pub_date": String,   // "20161204"
+      "title": String,      // "&lt;b&gt;성경&lt;/b&gt;"
+      "category": String,   // "종교 "
+      "pub_nm": String,     // "그라티아"
+      "isbn13": String,     // "9791187678014"
+    }
+    */
+    const book = {};
+    if (item.title && item.title !== '') book.title = KakaoBookAPIUtil.removeTag(item.title);
+    if (item.author && item.author !== '') book.author = KakaoBookAPIUtil.removeTag(item.author);
+    if (item.pub_nm && item.pub_nm !== '') book.publisher = KakaoBookAPIUtil.removeTag(item.pub_nm);
+    if (item.pub_date && item.pub_date !== '') book.publishedDate = item.pub_date;
+    if (item.category && item.category !== '') book.category = KakaoBookAPIUtil.removeTag(item.category);
+    if (item.isbn13 && item.isbn13 !== '') book.isbn = KakaoBookAPIUtil.removeTag(item.isbn13);
+    if (item.cover_s_url && item.cover_s_url !== '') book.thumbnail = item.cover_s_url;
+    if (item.cover_l_url && item.cover_l_url !== '') book.thumbnail = item.cover_l_url;
+
+    return book;
+  }
+
+  static removeTag(str) {
+    return str.replace(/&lt;b&gt;/g, '');
+  }
+
+
+  static pageParser(data, pageNo, result) {
+
+    if(data.channel.item.length === 0) {
+      return {
+        totalSize: data.channel.totalCount,
+        pageNo,
+        hasNext: false,
+        hasPrev: false
+      };
+    }
+
+    const totalSize = data.channel.totalCount;
+
+    // @Warning hasNext is only valid, when the result value is a constant while searching all the results.
+    const pageInfo = {
+      totalSize,
+      pageNo,
+      hasNext: totalSize > (pageNo - 1) * result + Number.parseInt(data.channel.result) && pageNo < 3
+    };
+    return pageInfo;
+  }
+}
+
```

Comment:
check out this lint. ( new line at the end of files. 
[https://github.com/eslint/eslint/blob/master/docs/rules/eol-last.md](https://github.com/eslint/eslint/blob/master/docs/rules/eol-last.md)
> Trailing newlines in non-empty files are a common UNIX idiom. Benefits of trailing newlines include the ability to concatenate or append to files as well as output files to the terminal without interfering with shell prompts.